k9mail-library/src/test/java/com/fsck/k9/mail/store/imap/ImapListTest.java

```diff
@@ -11,33 +14,129 @@
 import static org.junit.Assert.fail;
 
 public class ImapListTest {
-    @Test public void testImapListMethods() throws IOException {
+
+    private ImapList buildSampleList() {
         ImapList list = new ImapList();
         list.add("ONE");
         list.add("TWO");
         list.add("THREE");
+        return list;
+    }
+
+    @Test public void containsKey_returnsTrueForKeys() throws IOException {
+        ImapList list = buildSampleList();
 
         assertTrue(list.containsKey("ONE"));
         assertTrue(list.containsKey("TWO"));
         assertFalse(list.containsKey("THREE"));
         assertFalse(list.containsKey("nonexistent"));
```

Comment:
I'd get rid of the negative cases (the `assertFalse()` lines). They should be separate tests (that are actually already present further down).